packages/vitest/src/node/config.ts

```diff
@@ -124,30 +124,50 @@ export function resolveConfig(
   if (resolved.coverage.provider === 'v8' && resolved.coverage.enabled && isBrowserEnabled(resolved))
     throw new Error('@vitest/coverage-v8 does not work with --browser. Use @vitest/coverage-istanbul instead')
 
-  resolved.deps = resolved.deps || {}
+  resolved.deps ??= {}
+  resolved.deps.moduleDirectories ??= ['/node_modules/']
+  resolved.deps.moduleDirectories = resolved.deps.moduleDirectories.map((dir) => {
+    if (!dir.startsWith('/'))
+      dir = `/${dir}`
+    if (!dir.endsWith('/'))
+      dir += '/'
+    return normalize(dir)
+  })
+
+  resolved.server ??= {}
+  resolved.server.deps ??= {}
+
+  const deprecatedDepsOptions = ['inline', 'external', 'fallbackCJS'] as const
+  deprecatedDepsOptions.forEach((option) => {
+    if (resolved.deps[option] !== undefined) {
+      console.warn(
+        c.yellow(
+         `${c.inverse(c.yellow(' Vitest '))} "deps.${option}" is deprecated, use "server.deps.${option}" instead`,
```

Comment:
```suggestion
         `${c.inverse(c.yellow(' Vitest '))} "deps.${option}" is deprecated. If you rely on vite-node directly, use "server.deps.${option}" instead. Otherwise, consider using "optimizer.web.${options == 'external' ? 'exclude' : 'include'}"`,
```